mantle/platform/api/aliyun/api.go

```diff
@@ -128,14 +128,45 @@ func (a *API) CopyImage(source_id, dest_name, dest_region, dest_description, kms
 			Value: "mantle",
 		},
 	}
-
 	response, err := a.ecs.CopyImage(request)
 	if err != nil {
 		return "", fmt.Errorf("copying image: %v", err)
 	}
+
+	// if we have a need to operate on an image immediately after the image has
+	// been copied to a region, we can wait for it to be marked available
+	//
+	// NB: this gem from the Aliyun API docs - "A single region can have only
+	// one image copy task running at a time. Other image copy tasks queue up
+	// for the current task to complete before they run in sequence."
+	//
+	// Not sure if this means only one copy task can run for the *entire* region
+	// or if the limitation is per account, per region...but this kind of
+	// queuing would explain some of the delays observed.
+	if wait_for_ready {
+		a.WaitForImageReady(dest_region, response.ImageId)
+	}
 	return response.ImageId, nil
 }
 
+// WaitForImageReady checks that an image in a region is available to be
+// operated on. i.e. when you want to modify attributes of an image
+func (a *API) WaitForImageReady(region_id string, image_id string) error {
+	for {
```

Comment:
should we wait forever? If we wanted to timeout (even if it is a high timeout) we could use the new `RetryUntilTimeout()` function: 363a1b0

Reply:
Ehhh...we already block indefinitely in `ImportImage()` (though I have never seen that take terribly log), so I was modeling that behavior.

I'll tinker with the retry function